components/style/properties/properties.mako.rs

```diff
@@ -2560,6 +2576,10 @@ impl<'a> StyleBuilder<'a> {
             device,
             parent_style,
             inherited_style,
+            // None of our callers pass in ::first-line parent styles.
```

Comment:
Well, we always pass in the parent element style.

Whether this _does_ the right thing when first-line is involved, I'm not sure.  Will need to test.